concordium-node/src/consensus_ffi/ffi.rs

```diff
@@ -709,14 +714,37 @@ impl ConsensusContainer {
         ))
     }
 
-    pub fn in_baking_committee(&self) -> ConsensusIsInBakingCommitteeResponse {
-        wrap_c_committee_call!(self, |consensus| bakerIdBestBlock(consensus))
+    pub fn in_baking_committee(&self) -> (ConsensusIsInBakingCommitteeResponse, Option<u64>) {
+        let consensus = self.consensus.load(Ordering::SeqCst);
+
+        unsafe {
+            let mut baker_id: u64 = 0;
+            let mut has_baker_id: u8 = 0;
+
+            let result = bakerStatusBestBlock(consensus, &mut baker_id, &mut has_baker_id);
```

Comment:
Only the foreign call needs to be unsafe. We should strive to have the smallest possible scope for the unsafe blocks.